sdk/iot/provisioning/src/az_iot_provisioning_client.c

```diff
@@ -47,161 +54,420 @@ AZ_NODISCARD az_result az_iot_provisioning_client_init(
 }
 
 // <id_scope>/registrations/<registration_id>/api-version=<service_version>
-AZ_NODISCARD az_result az_iot_provisioning_client_user_name_get(
+AZ_NODISCARD az_result az_iot_provisioning_client_get_user_name(
     az_iot_provisioning_client const* client,
-    az_span mqtt_user_name,
-    az_span* out_mqtt_user_name)
+    char* mqtt_user_name,
+    size_t mqtt_user_name_size,
+    size_t* out_mqtt_user_name_length)
 {
   AZ_PRECONDITION_NOT_NULL(client);
-  AZ_PRECONDITION_VALID_SPAN(mqtt_user_name, 0, false);
-  AZ_PRECONDITION_NOT_NULL(out_mqtt_user_name);
+  AZ_PRECONDITION_NOT_NULL(mqtt_user_name);
```

Comment:
we should have a not null or empty?

Reply:
since it's a `char*` now we check for not `NULL` here and the size is greater than one on the next line.